### ISSUE TITLE
Azure Kubernetes Service extension can't connect to AKS clusters with same name but are in different resource groups 

### ISSUE THREAD

**Issue by SOFSPEEL** [Azure/vscode-aks-tools#441]
**Describe the bug**
Have 2 clusters with the **same name** in the same subscription but in different resource groups
This works in version 1.3.18 and earlier version of **Azure Kubernetes Service** Extension, so this was broken in 1.4.0 version
e.g. 
Subscription: ACS-MGMT-DEV01

cluster 1: mobileservice
resource group: **eu1d**-mobileservice-aks

cluster 2: mobileservice
resource group: **eu1t**-mobileservice-aks

Note that the tool does not connect successfully to the second cluster and shows "invalid"

**Mention the platform you are using**
vscode
Version: 1.85.1 (Universal)
Commit: 0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2
Date: 2023-12-13T09:48:06.308Z
Electron: 25.9.7
ElectronBuildId: 25551756
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 23.2.0

**To Reproduce**
1. Go to 'Kubernetes' CLOUDS/Azure section in vscode
2. Expand to  an Azure subscription that has multiples AKS with same name but are in different resource groups
3. Results in the cluster name 'invalid' for all but the first AKS cluster.

**Expected behavior**
Should connect to both clusters.

**Screenshots**
<img width="218" alt="image" src="https://github.com/Azure/vscode-aks-tools/assets/1612646/260a3f04-1757-4a20-a101-a6eb63102a10">

**Desktop (please complete the following information):**
```
  Model Name:	MacBook Pro
  Model Identifier:	Mac14,5
  Model Number:	MPHG3LL/A
  Chip:	Apple M2 Max
  Total Number of Cores:	12 (8 performance and 4 efficiency)
  Memory:	32 GB
  System Firmware Version:	10151.61.4
  OS Loader Version:	10151.61.4
```

**Comment by Tatsinnit** [comment]
Thank you so much @SOFSPEEL for initiating this. ☕️❤️  I can readily replicate the steps provided, and I am exploring potential solutions. We will carefully examine the issue and strive to devise the best possible solution to address it effortlessly. Thank you for sharing; I'll be in touch soon.

Gentle fyi & cc to some peers: @peterbom, @hsubramanianaks, @sprab & @qpetraroia for eyes and fyi. (I might have solution for this already and will open PR by tomorrow for your eyes) 🤹‍♀️🙏

**Comment by Tatsinnit** [comment]
Hiya @SOFSPEEL, looping back to inform that the new released version for aks extension`1.4.1` contains fix for this. Thanks heaps.
